fluent-package/msi/assets/fluentd.bat

```diff
@@ -1,22 +1,26 @@
 @echo off
 if "%~nx0" == "td-agent.bat" (
   set FLUENT_PACKAGE_TOPDIR=%~dp0..\
-  set TD_AGENT_TOPDIR=%~dp0..\..\td-agent
 ) else (
   set FLUENT_PACKAGE_TOPDIR=%~dp0
-  set TD_AGENT_TOPDIR=%~dp0..\td-agent
 )
+
+@rem Convert path separator from backslash to forwardslash
+setlocal enabledelayedexpansion
+set FLUENT_PACKAGE_TOPDIR="FLUENT_PACKAGE_TOPDIR=!FLUENT_PACKAGE_TOPDIR:\=/!"
+endlocal

```

Comment:
Somehow, this does not appear to work for

* https://github.com/fluent/fluentd/issues/1138

I tested the package with the following config, but backslash was still used and could not load the existing buffer files on restart.

Maybe we should handle this separately from this PR.

```xml
<source>
  @type sample
  tag test
</source>

<match test.**>
  @type stdout
  <buffer>
    @type file
    path "#{ENV['FLUENT_PACKAGE_TOPDIR']}/var/log/fluentd/buffer/stdout"
    flush_at_shutdown false
    flush_interval 10
  </buffer>
</match>
```

log

```
[info]: using configuration file: <ROOT>
  <source>
    @type sample
    tag "test"
  </source>
  <match test.**>
    @type stdout
    <buffer>
      @type "file"
      path "C:\\opt\\fluent\\/var/log/fluentd/buffer/stdout"
      flush_at_shutdown false
      flush_interval 10
    </buffer>
  </match>
</ROOT>
```

Reply:
it will be fixed via ab9df59

Reply:
Here is the example to verify scope of enabledelayedexpansion:

```
set FLUENT_PACKAGE_TOPDIR=%~dp0..\
set TD_AGENT_TOPDIR=%~dp0..\td-agent
setlocal enabledelayedexpansion
echo "FLUENT_PACKAGE_TOPDIR:"
set FLUENT_PACKAGE_TOPDIR=!FLUENT_PACKAGE_TOPDIR:\=/!
set TD_AGENT_TOPDIR=!TD_AGENT_TOPDIR:\=/!
echo %TD_AGENT_TOPDIR%
echo %FLUENT_PACKAGE_TOPDIR%
endlocal

echo "OUTSIDE"
echo %TD_AGENT_TOPDIR%
echo %FLUENT_PACKAGE_TOPDIR%
```

![expansion](https://github.com/fluent/fluent-package-builder/assets/225841/d0525482-95f5-4d37-864d-3c1ad9c3031c)

Reply:
Wow! I see! Thanks for the fix!

Reply:
I will check it after the CI finished and the new package is built.

Reply:
https://github.com/fluent/fluent-package-builder/pull/494#discussion_r1259005439 seems still not solved.
I checked the package, but the result is the same.

Sorry, there are some points I don't understand.

https://github.com/fluent/fluent-package-builder/pull/494#discussion_r1259179257 means the slash conversion is lost outside the scope?
If so, the environmental variable `ENV['FLUENT_PACKAGE_TOPDIR']` in the config should still contain backslashes?

I checked the previous environmental variable `ENV['TD_AGENT_TOPDIR']` in the config.
It returns `C:\\opt\\td-agent\\`.
I think it is strange...
Now it seems to me that no one sets this old variable anymore.
Why can Fluentd still access this variable...

I think it is correct that this old environment variable still works as is.
If not, then the old config using this variable can't work in v5.
But I don't understand how this still works now...

Reply:
Summary

* `ENV['TD_AGENT_TOPDIR']` should work in the new package, but it is uncertain in the current implementation.
* The logic converting BachSlash in `ENV['FLUENT_PACKAGE_TOPDIR']` works for the environmental variables, but it would not be used from the configuration.

We should handle these points in other PRs.